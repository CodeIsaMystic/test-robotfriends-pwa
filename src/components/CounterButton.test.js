import React from 'react';
import { shallow } from 'enzyme';

import CounterButton from './CounterButton';





/*   Entering all tests in describe   */
describe('counterButtonw', () => {

  //const counterButton = shallow(<CounterButton />);

  it('expect to render counter button component', () => {
    const mockColor = 'red';

    expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
  });



  it('correctly increments the counter', () => {
    const mockColor = 'red';
    const wrapper = shallow(<CounterButton color={mockColor} />);

    expect(wrapper).toMatchSnapshot();


    /*  Simulating click event then test if its fail or not
    and look up what s expected */
    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({ count: 2 })


    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({ count: 3 })


    //expect(wrapper.props()).toEqual({ color: 'red' })
    expect(wrapper.props().color).toEqual('red')
  });

});