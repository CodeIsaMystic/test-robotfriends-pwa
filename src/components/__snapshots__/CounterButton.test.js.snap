// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly increments the counter 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;

exports[`expect to render counter button component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
